dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/PageParams.java

```diff
@@ -55,13 +56,21 @@ public class PageParams {
   /** Indicates whether to fetch the total number of items. */
   final boolean pageTotal;
 
-  public PageParams(Integer page, Integer pageSize, boolean pageTotal) {
+  private PageParams(Integer page, Integer pageSize, boolean pageTotal) {
     this.page = Objects.requireNonNullElse(page, DEFAULT_PAGE);
     this.pageSize = Objects.requireNonNullElse(pageSize, DEFAULT_PAGE_SIZE);
     this.pageTotal = pageTotal;
   }
 
-  public static PageParams of(Integer page, Integer pageSize, boolean pageTotal) {
+  public static PageParams of(Integer page, Integer pageSize, boolean pageTotal)
+      throws BadRequestException {
+    if (page != null && page < 1) {
```

Comment:
move that validation into the constructor

Reply:
isn't the point of using a factory to run the validations within it?
if anything, I think we should move the validations in the constructor here. Would you agree?